### ISSUE TITLE
macOS: Undecorated windows ignores resizable option

### ISSUE THREAD

**Issue by Andos** [glfw/glfw#902]
macOS is perfectly capable of resizing undecorated windows but GLFW only sets the resizable mask in _"getStyleMask"_ in _cocoa_window.m_ inside the 'else' part of the test for the window decoration flag.

If I move out 
```
	if (window->resizable)
		styleMask |= NSWindowStyleMaskResizable;
```
till after the check I get the desired behavior or being able to resize my undecorated windows.

I guess it is a question of consistency across platforms. The GLFW documentation sais nothing about the GLFW_DECORATED flag affecting the resize behavior of windows though.

**Comment by elmindreda** [comment]
Yup, it's a consistency thing. I'll update the documentation.

Do you need resizable undecorated windows on macOS, and if so, do you mind if I ask for what purpose?

**Comment by Andos** [comment]
Yeah It is something I need for completely custom drawn windows. Think UI's like Photoshop where they draw everything themselves except the popup menus.

I'm drawing a title bar myself based and use this method in NSWindow to perform window dragging:
https://developer.apple.com/reference/appkit/nswindow/1419386-performwindowdragwithevent?language=objc (available from El Capitain 10.11)

Window resizing is something that I wouldn't want to implement myself as the resize handle area could be outside the window.

**Comment by elmindreda** [comment]
I have to ask you to keep this modification out-of-tree for the time being. Will fix this in the next revision of the API so it works as expected.

**Comment by elmindreda** [comment]
The documentation has been updated with 29b72c59f3fb35238300b860acc646aa5764cad7 for now.
